### DISCUSSION TITLE
Why do I always get 502?

### DISCUSSION THREAD

**Discussion by rowild** [codesandbox/codesandbox-client#6656] [Q&A]
This sandbox:
https://codesandbox.io/s/quasar2vue3setup-forked-u2siof
(first linked to a github repo, then forked)
only yields 502. I have no idea why. Locally it works perfect. 

Any ideas?

**Comment by JamesACS** [comment]
Hey @rowild - I'm seeing this error appear: 

error @quasar/app-vite@1.0.0-beta.14: The engine "node" is incompatible with this module. Expected version "^18 || ^16 || ^14.19". Got "14.18.1"
error Found incompatible module.

Where did you fork the sandbox from?

**Comment by rowild** [comment]
Thank you for our feedback, @JamesACS !

It's been a while - I didn't fork the repo, but create it on my own, commit it to github (were I can't find it anymore) and then "forked" (copied it over) to codesandbox.

Do you happen to know, why a Node version is selected that does not fit? The quasar.config.js file clearly states that a Node v16 should be used:

```js
   build: {
      target: {
        browser: ['es2019', 'edge88', 'firefox78', 'chrome87', 'safari13.1'],
        node: 'node16'
      },
  ```

Also, changing the node version in package.json does not improve the situation...

**Reply by rowild** [reply]
It's the @quasar/app-vite dependency that explicitly demands at least node 14.19:
https://github.com/quasarframework/quasar/blob/dev/app-vite/package.json
But I do not find the reason, why my app uses Node 14.18.1...

**Reply by rowild** [reply]
Even changing the sandbox.config.js to use node 16 doesn't work:

```json
{
  "infiniteLoopProtection": true,
  "hardReloadOnChange": false,
  "view": "browser",
  "container": {
    "node": "16",
    "port": 8055
  }
}
```

**Reply by rowild** [reply]
OK, I removed the v14 dependency from package.json, and now yarn at least installs the dependencies and can save a lockfile. However, the 502 status remains...

**Reply by lbogdan** [reply]
You don't have any "start" script defined; by default we're looking for `dev`, `develop`, `serve` or `start`, in this order, and we just give up if we don't find any.

**Reply by rowild** [reply]
Aaaaaahhhh!!!!! @lbogdan thank you so much!!!

**Reply by rowild** [reply]
Ok, I was a little bit too quick with my happiness. Quasar apps start with `quasar dev` using the quasar cli. So I installed `yarn global add @quasar/cli` via a separate terminal window. Then I added `"dev": "npx quasar dev"` to package.json, which is indeed called. But the 520 remains. 
@lbogdan Any idea what I am still doing wrong? Does codesandbox redirect localhost:9000 in a "useful was"? (sorry for my poor formulation...)

**Reply by lbogdan** [reply]
I made it work here: https://codesandbox.io/s/quasar2vue3setup-forked-t3gne2 . What I changed:

- removed the port from `sandbox.config.json` (as it was different from Quasar's 9000)

- in `quasar.config.js`:

```
[...]
    devServer: {
      https: false,
      open: false // opens browser window automatically
    },
[...]
```
, as we don't support `https` servers. Also `open` would only work locally, but this is a bit unrelated.

**Reply by rowild** [reply]
@lbogdan I really appreciate your support!! Thank you very much!

I applied the changes to my setup, but now new errors appear. First there is a HTML parsing error: HTML comment `<!--` cannot be parsed correctly. This does not happen in your fork - but I haven't found out why. In my local VS Code setup a VS Code extension Volar would take of that, but I don't know how to install such an extension on codesandbox (if that is possible at all).

Second: If I remove the HTML comments, the `@before-appear...` bindings on the `Transition` component throw an error. This error is not shown as red underlined in our fork, but they show up in the console.

May I kindly ask, if you happen to have any more insight into these problems?

**Reply by rowild** [reply]
Here are screenshots:

![Bildschirmfoto 2022-05-03 um 20 38 33](https://user-images.githubusercontent.com/213803/166521258-60b4a6f1-3e75-48ce-86ee-2fed2e98a691.png)

![Uploading Bildschirmfoto 2022-05-03 um 20.38.43.png…]()

**Reply by rowild** [reply]
It also seems that `<script setup>` is recognised as JSX code...

**Reply by JamesACS** [reply]
Hey @rowild - I'm not seeing any issues now - did you manage to resolve the problem?

**Reply by rowild** [reply]
@JamesACS Thank you for asking, but no. The squiggly lines, as shown in the previous screenshot, are still there. No idea how to fix that. Do you have an idea?

**Reply by JamesACS** [reply]
@rowild - I've been looking into this but haven't found a solution, the error is being thrown by our ESLint integration but doesn't seem to affect functionality.
